### PR TITLE
[AEA-1264] connection protocol restrictions check by multiplexer

### LINKED ISSUE #2209: Refactor restricted/excluded protocols check for connections
**Is your feature request related to a problem? Please describe.**
Looks like every connection performs check for protocols support by it's own.

**Describe the solution you'd like**
The code is mostly the same and would be nice to refactor connecetions to use shared code for this operation.
It can be done with multiplexer check before send operation/on connection base class.

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

connection protocol restrictions check by multiplexer

## Fixes

#2209

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/mail/base.py [hunk 1]
```diff
@@ -209,7 +209,7 @@ def _get_public_ids_from_uri(
             )
         return (skill_id, connection_id)
 
-    def __str__(self):
+    def __str__(self):  # pragma: nocover
```
Comment (DavidMinarsch) [inline aea/mail/base.py:212]: This should have a test. These methods can easily break.

File: aea/protocols/base.py [hunk 2]
```diff
@@ -424,4 +424,4 @@ def protocol_specification_id(self) -> PublicId:
 
     def __repr__(self) -> str:
         """Get str repr of the protocol."""
-        return f"Protocol({self.protocol_id})"
+        return f"Protocol({self.protocol_id})"  # pragma: nocover
```
Comment (DavidMinarsch) [inline aea/protocols/base.py:427]: see above

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Review by marcofavorito — APPROVED** [review]
LGTM

File: aea/protocols/base.py (hunk 2 above)
Comment (solarw) [reply aea/protocols/base.py:427]: done

File: aea/mail/base.py (hunk 1 above)
Comment (solarw) [reply aea/mail/base.py]: done

**Pull request merged**